### PR TITLE
Pashah/subscriptions

### PR DESCRIPTION
**Pull request by parthsha**
Changes proposed in this pull request:
 - Rename Set-RsSubscription to Copy-RsSubscription as it really just made a copy of an existing subscription.
 - Rename Update-RsSubscription to Set-RsSubscription as all other update commands are called Set-*.
 - Added capability to specify FileShareCredentials, Subject, BCC among other things in New-RsSubscription.

How to test this code:
 - Updated tests to reflect changes made to each command.
 - Added new tests to cover any added functionality.

Has been tested on (remove any that don't apply):
 - Powershell 5
 - Windows 10
 - Power BI Report Server (October 2017)

### CODE REVIEW

**Comment by parthsha** [comment]
@warren2600, @markwragg, @ClaudioESSilva, @jzabroski:  Can you please tell me what you feel about the changes proposed in this Pull Request?

**Comment by markwragg** [comment]
Makes sense to me! Obviously a breaking change but I think the names are now more logical.

**Comment by parthsha** [comment]
Thanks for the feedback, @markwragg!

File: ReportingServicesTools/Functions/CatalogItems/Set-RsSubscription.ps1 [hunk 1]
```diff
@@ -1,154 +1,142 @@
...
-            A object with all subscritpion configurations. The default output of Get-RsSubscription. You must piping it to this command.
-        
         .PARAMETER ReportServerUri
             Specify the Report Server URL to your SQL Server Reporting Services Instance.
             Use the "Connect-RsReportServer" function to set/update a default value.
-        
+
         .PARAMETER Credential
-            Specify the credentials to use when connecting to the Report Server.
+            Specify the password to use when connecting to your SQL Server Reporting Services Instance.
```
Comment (ClaudioESSilva) [inline ReportingServicesTools/Functions/CatalogItems/Set-RsSubscription.ps1]: To keep the consistency, please change from "password" to "credentials". AFAIK this is the same type the other credentials.

**Comment by parthsha** [comment]
Thank you for the feedback @ClaudioESSilva! I will release this as 0.0.4.0 as it does contain quite a lot of changes (Commands for managing Embedded Data Sources for Paginated/Power BI Reports and Subscriptions).

**Comment by jzabroski** [comment]
This is great and I am appreciative you tagged me. I'll have down time next
week to upgrade from 0.2.9-fork to 0.4. Hopefully will contribute more too.

**Comment by warren2600** [comment]
Excellent changes to make function names more consistent and clear. Great work!

**Pull request merged**